src/theme-i18n.json

```diff
@@ -0,0 +1,51 @@
+{
```

Comment:
Ideally, wp-cli would take this file from the WordPress core if it exists. If it doesn't, it shouldn't lookup for strings in the directory, as it means that the WordPress version in use doesn't support `theme.json`.

Reply:
Note that `wp i18n make-pot` is usually run against a single theme's directory, with no knowledge of a WP installation around it.

Ideally the `theme-i18n.json` file is requested from an external API, e.g. WordPress.org, based on the `version` field in the `theme.json` file.

It shouldn't be bundled with WP-CLI for sure.

BTW, the same would need to be done for `block.json`.

Reply:
As per https://github.com/wp-cli/i18n-command/issues/224#issuecomment-736646832 by @schlessera my understanding is that we want to access this from WordPress core. What would be the preferred way to do so?

Reply:
(same as below, didn't see Pascal's comment when posting mine because I hadn't reloaded the browser)

Reply:
It may take a bit to create and deploy an endpoint in wordpress.org. Would you think it's reasonable to bundle this file with 2.5.1 so we then can update translate.wordpress.org for these strings to be translatable for the WordPress 5.8 release? We can work on that endpoint later, with more time.

Reply:
> BTW, the same would need to be done for block.json.

I don't know what this means. Is the idea to change the implementation of the existing `BlockExtractor.php` so it takes its data from a similar file?

Reply:
> I don't know what this means. Is the idea to change the implementation of the existing BlockExtractor.php so it takes its data from a similar file?

Yes. This file already exists in Gutenberg: https://github.com/WordPress/gutenberg/blob/77bc9cae4d016f52523f0ee7018183961ab6056e/packages/blocks/src/api/i18n-block.json

Reply:
I want to share an alternative proposal: what if we have a PHP package that bundles this logic (code + `theme-i18n.json`) and can be reused across wp-cli, WordPress core, and Gutenberg?

It'd have a few advantages: 1) no need to be online to create a `.pot` file, run WordPress or the plugin; 2) modifications to the `theme-i18n.json` structure that affect the code can be distributed from a single point instead of having to coordinate updates to three codebases.

This would take some time to figure out, so I'd like to propose that we bundle `theme-i18n.json` with this command for the sake of releasing 2.5.1.

Reply:
Ideally, this file would be fetched online on-demand, with a committed version in this repo being the fallback if no online access or the download failed (which would also show a warning in that case).
The committed fallback here could be kept in sync via a GitHub action, so that we're eventually up-to-date without manual intervention.

Reply:
What we'd need is a reliable URL where this file can be fetched in its latest version. It doesn't necessarily have to be an API, but it should be as stable as possible.

Reply:
In https://github.com/wp-cli/i18n-command/pull/254/commits/6d850a120f7e6a24fb71a5e7091228497befde32 I'm pulling the file from Gutenberg's GitHub repo. I've also added a fallback to the local file if there was an issue and it is empty. I've tried using WordPress.org SVN but it reports a 403.

If this unblocks us to ship this, let's do it. My preference, however, would be to start only with the local file as it less fragile.

Reply:
Added some stress testing instructions in the description.

Reply:
I agree with @nosolosw that for the first version a bundled schema file should only be used to not depend on an unstable URL like it's implemented right now. It's unstable because there's no version at all. In case there's a change to the structure of the file or the file is moved we'd have to update WP-CLI anyway right?
I don't think it's responsible to block/delay the command from working in this case just because a not yet published update isn't supported.

A compromise might be to add a `--prefer-local-schema` argument which can be used by environments which require a stable behaviour?

Reply:
@schlessera Any thoughts on a `--prefer-local-schema` argument?

Reply:
@ocean90 Sounds useful, especially if it accepts a file path (which would default to the included one). This would let you easily experiment with different schema.